html/captive-portal/lib/captiveportal/PacketFence/Controller/Status.pm

```diff
@@ -96,6 +107,18 @@ sub userIsAuthenticated : Private {
     $c->stash(
         nodes    => \@nodes,
     );
+    if (is_dpsk_enabled($c->profile)) {
+        my @provisioners = $c->profile->provisionerObjects();
+        unless(@provisioners){
+            return;
+        }
+
+        foreach my $provisioner (@provisioners) {
+            next unless $provisioner->isa("pf::provisioner::dpsk");
+            my $dpsk = $provisioner->generate_dpsk($pid);
```

Comment:
If there is more than one, it's going to generate it each time but only one item is in the stash after.
Will it behave correctly if more than one DPSK provisioner is setup on the connection profile?

Reply:
In fact the better option is probably to put the dpsk in the stack if the dpsk exist in the person field.
If psk doesnt exist for the specific user then put an empty dpsk in the stash.